distributed/semaphore.py

```diff
@@ -460,6 +462,26 @@ def acquire(self, timeout=None):
         timeout = parse_timedelta(timeout)
         return self.client.sync(self._acquire, timeout=timeout)
 
+    async def _release(self):
+        # popleft to release the oldest lease first
+        lease_id = self._leases.popleft()
+        logger.info("%s releases %s for %s", self.client.id, lease_id, self.name)
+
+        try:
+            return await retry_operation(
+                self.client.scheduler.semaphore_release,
+                name=self.name,
+                lease_id=lease_id,
+                operation="semaphore release: client=%s, lease_id=%s, name=%s"
+                % (self.client.id, lease_id, self.name),
+            )
+        except OSError:  # Too many broken connections, release fails
+            logger.error(
+                "Release failed for client=%s, lease_id=%s, name=%s. Cluster network might be unstable."
+                % (self.client.id, lease_id, self.name),
+            )
+            return False
```

Comment:
The release itself doesn't return anything. The `False` return value is not used anywhere. Why introduce this?

Reply:
At the very least I would then return a True if it was released successfully

Reply:
> Why introduce this?

So that a user/machine can easily figure out if the release did not succeed

Reply:
> At the very least I would then return a True if it was released successfully

I agree, this way we keep the logic consistent. I will change the appropriate tests then, unless you suggest a better alternative